### PR TITLE
Refactor properties documentation

### PR DESCRIPTION
**Pull request by lmazuel**
Fix https://github.com/Azure/azure-sdk-for-python/issues/8199

This PR enhance the base class documentation and remove the redondancy is all subclasses.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py [hunk 1]
```diff
@@ -112,28 +112,62 @@ def __exit__(self, *args):
...
         return self._format_url(self._hosts[LocationMode.SECONDARY])
 
     @property
     def secondary_hostname(self):
+        """The hostname of the secondary endpoint.
+
+        If not available this will be None. To explicitly specify a secondary hostname, use the optional
+        `secondary_hostname` keyword argument on instantiation.
+
+        :type: str
```
Comment (bryevdv) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: or None?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py [hunk 2]
```diff
@@ -112,28 +112,62 @@ def __exit__(self, *args):
 
     @property
     def url(self):
+        """The full endpoint URL to this entity, including SAS token if used.
+
+        This could be either the primary endpoint,
```
Comment (bryevdv) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: nit: (here and elsewhere) line is unnecessarily short

Comment (lmazuel) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: for this one, I prefer personally to cut at the comma :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py (hunk 1 above)
Comment (lmazuel) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: fixed

**Pull request merged**